### PR TITLE
Annotate tsc output on lint

### LINKED ISSUE #553: Support Buildkite annotations for lint

### PR DESCRIPTION
**Pull request by 72636c**
Closes #553 for realsies this time.

This also tentatively fixes out-of-order stdout logs.

### CODE REVIEW

File: src/cli/adapter/prettier.ts [hunk 1]
```diff
@@ -172,7 +171,7 @@ export const runPrettier = async (
   if (result.errored.length) {
     logger.plain(`Flagged ${logger.pluralise(result.errored.length, 'file')}:`);
     for (const { err, filepath } of result.errored) {
-      logger.warn(filepath, ...(err ? [inspect(err)] : []));
+      logger.warn(filepath, ...(err ? [String(err)] : []));
```
Comment (72636c) [inline src/cli/adapter/prettier.ts:174]: We get good-enough error messages without a full-blown `inspect`, which enumerates additional properties including a verbose stack trace.

File: src/utils/exec.ts [hunk 2]
```diff
@@ -169,24 +168,14 @@ export const execConcurrently = async (
...
-        (a) =>
-          `[${a.command.name}] ${chalk.bold(
-            a.command.command,
-          )} exited with code ${chalk.bold(a.exitCode)}`,
-      );
-
-    log.newline();
-
-    messages.forEach((message) => log.err(message));
-    log.newline();
```
Comment (72636c) [inline src/utils/exec.ts:185]: This was also unnecessary as each message is already logged by concurrently.

File: src/cli/lint/external.ts [hunk 3]
```diff
@@ -32,24 +50,32 @@ const lintSerially = async (input: Input) => {
...
+  tscOutputStream.pipe(input.tscOutputStream ?? process.stdout);
+
+  const { eslint, prettier, tscOk } = await lint({ ...input, tscOutputStream });
 
   if (eslint.ok && prettier.ok && tscOk) {
     return;
   }
 
+  if (!isSerial) {
+    // stdio from worker threads can lag a bit, so we wait for a little while.
```
Comment (etaoins) [inline src/cli/lint/external.ts:68]: This is annoying me but I can't think of anywhere we're missing a flush or an `await`. Is this just data bouncing around streams and `postMessage` that haven't completely settled yet?

**Review by etaoins — APPROVED** [review]

File: src/cli/lint/external.ts (hunk 3 above)
Comment (72636c) [reply src/cli/lint/external.ts]: Thanks for calling me on this, I think I've fixed it proper in de883b05935975a73f45bdb92b01815433469d45.

**Pull request merged**